code/modules/antagonists/_common/antag_helpers.dm

```diff
@@ -7,3 +7,35 @@
 			continue
 		if(!antag_type || !specific && istype(A,antag_type) || specific && A.type == antag_type)
 			. += A.owner
+
+/// From a list of players (minds, mobs or clients), finds the one with the highest playtime (either from a specific role or overall living) and returns it.
+/proc/get_most_experienced(list/players, specific_role)
+	if(!CONFIG_GET(flag/use_exp_tracking)) //woops
+		return players[1]
+	var/index = 0
+	for(var/client/player as anything in players)
+		index++
+		player = get_player_client(player)
+		if(!player?.prefs)
+			continue
+		if(!length(player.prefs.exp))
+			player.set_exp_from_db()
+			if(!length(player.prefs.exp))
+				continue
+		var/client/most_played = get_player_client(players[1])
+		if(player == most_played)
+			continue
+		if(!most_played.prefs || !length(most_played.prefs.exp))
+			players.Swap(1, index)
+			continue
+		var/player_playtime
+		var/most_playtime
+		if(specific_role)
+			player_playtime = player.prefs.exp[specific_role] ? text2num(player.prefs.exp[specific_role]) : 0
+			most_playtime = most_played.prefs.exp[specific_role] ? text2num(most_played.prefs.exp[specific_role]) : 0
+		else
+			player_playtime = player.get_exp_living(TRUE)
+			most_playtime = most_played.get_exp_living(TRUE)
+		if(player_playtime > most_playtime)
+			players.Swap(1, index)
+	return players[1]
```

Comment:
There is a much easier way to do this:
Have a variable that is the person with the most exp.
If it isn't defined, set it to the player. Otherwise, compare their exp with the next player. If the next player surpasses the person with the most exp, set them to the person with the most exp.
You don't need a bunch of these variables and list manipulations as a result.